distribution/docs/src/main/resources/_contents/_developing/developing-intro-contents.adoc

```diff
@@ -0,0 +1,816 @@
+. Select *Java → Editor → Save Actions*.
+. Select *Perform the selected actions on save*.
+. Select *Format source code*.
+. Select *Format all lines* or *Format edited lines*, as necessary.
+. Optionally, select *Organize imports* (recommended).
+. Select the *Apply* button.
+. Select the *OK* button.
+
+===== Format Source Code Using IntelliJ
+
+In the toolbar, select *Code → Reformat Code* or use the keyboard shortcut `Ctrl-Alt-L`.
+
+==== ${branding} Software Versioning
+
+${branding} follows the https://www.osgi.org/wp-content/uploads/SemanticVersioning.pdf[Semantic Versioning White Paper] for bundle versioning.
+
+==== Ensuring Compatibility
+
+===== Compatibility Goals
+
+The ${branding} framework, like all software, will mature over time.
+Changes will be made to improve efficiency, add features, and fix bugs.
+To ensure that components built for ${branding} and its sub-frameworks are compatible, developers must use caution when establishing dependencies from developed components.
+
+===== Guidelines for Maintaining Compatibility
+
+====== ${branding} Framework
+
+For components written at the ${branding} Framework level (see Developing at the Framework Level), adhere to the following specifications:
+
+[cols="2,1,5" options="header"]
+|===
+
+|Standard/Specification
+|Version
+|Current Implementation (subject to change)
+
+|OSGi Framework
+|4.2
+|Apache Karaf 2.x +
+Eclipse Equinox
+
+|OSGi Enterprise Specification
+|4.2
+|Apache Aries (Blueprint) +
+Apache Felix `FileInstall` and `ConfigurationAdmin`
+|===
+
+[IMPORTANT]
+====
+Avoid developing dependencies on the implementations directly, as compatibility in future releases is not guaranteed.
+====
+
+==== ${branding} Catalog API
+
+For components written for the ${branding} Catalog (see Developing Catalog Components), only dependencies on the current major version of the Catalog API should be used.
+Detailed documentation of the Catalog API can be found in the Catalog API Javadocs.
+
+[cols="1,1,5"]
+|===
+
+|Dependency
+|Version Interval
+|Notes
+
+|${branding} Catalog API
+|[2.0, 3.0)
+|Major version will be incremented (to 3.0) if/when compatibility is broken with the 2.x API.
+
+|===
+
+=== Development Recommendations
+
+==== Javascript
+
+Avoid using `console.log`
+
+==== Package Names
+
+Use singular package names.
+
+==== Author Tags
+
+Author tags are discouraged from being placed in the source code, as they can be a barrier to collaboration and have potential legal ramifications.
+
+==== Unit Testing
+
+All code should contain unit tests that are able to test out any localized functionality within that class.
+When working with OSGi, code may have references to various services and other areas that are not available at compile-time.
+One way to work around the issue of these external dependencies is to use a mocking framework.
+
+.Recommended Framework
+[NOTE]
+====
+The recommended framework to use with ${branding} is https://github.com/mockito/mockito[Mockito].
```

Comment:
We may want to mention PowerMock for testing static methods and also Spock as an alternate testing/mocking framework as it is also a viable alternative.